2020/day15/solution.py

```diff
@@ -0,0 +1,28 @@
+from __future__ import  annotations
+
+input = [15,5,1,4,7,0]
+
+def find_nth_number(starting : list[int], nth : int) -> int:
+    last_spoken = { value : [index] for index, value in enumerate(starting)}
+    round = len(starting)
+    last_number = starting[-1]
+    while round < nth:
+        if len(last_spoken[last_number]) == 1:
```

Comment:
I had a good amount of fun golfing the logic, mainly focused on recusing the number of branches. Things I did:
- Unify the length 1 & length 2 case by changing the representation of length 1 lists `[round]` to instead be `[None, round]`, and using an inline ternary to handle the None. I think this one is reasonable to do and lets you avoid an `OOPS` case entirely - always the same format.
- Use a defaultdict to avoid having to explicitly create empy lists. Also seems reasonable.
- Changed the loop to range from 0 to `nth`, and moved the addition of the starting data to `last_spoken` into the loop. Dunno if this would make sense in reality or it's just code golfing gone too far :upside_down_face:

Reply:
yea, i  don't love it. I'm  actually  looking at rewriting it, because I think you really only need the last one if you are explicit about tracking the gap rather than the last two occurrences to compute.  let me see if i can get that working.

I am also looking at moving the starting into the loop as well